modules/chanserv/count.c

```diff
@@ -75,17 +80,31 @@ cs_cmd_count(struct sourceinfo *si, int parc, char *parv[])
 		else
 			othercnt++;
 	}
-	if (ca_hop == ca_vop)
-		command_success_nodata(si, _("%s: VOP: %u, AOP: %u, SOP: %u, AKICK: %u, Other: %u"),
-				chan, vopcnt, aopcnt, sopcnt, akickcnt, othercnt);
+	if (show_akicks)
+	{
+		if (ca_hop == ca_vop)
+			command_success_nodata(si, _("%s: VOP: %u, AOP: %u, SOP: %u, AKICK: %u, Other: %u"),
+					chan, vopcnt, aopcnt, sopcnt, akickcnt, othercnt);
+		else
+			command_success_nodata(si, _("%s: VOP: %u, HOP: %u, AOP: %u, SOP: %u, AKICK: %u, Other: %u"),
+					chan, vopcnt, hopcnt, aopcnt, sopcnt, akickcnt, othercnt);
+	}
 	else
-		command_success_nodata(si, _("%s: VOP: %u, HOP: %u, AOP: %u, SOP: %u, AKICK: %u, Other: %u"),
-				chan, vopcnt, hopcnt, aopcnt, sopcnt, akickcnt, othercnt);
+	{
+		if (ca_hop == ca_vop)
+			command_success_nodata(si, _("%s: VOP: %u, AOP: %u, SOP: %u, Other: %u"),
+					chan, vopcnt, aopcnt, sopcnt, othercnt);
+		else
+			command_success_nodata(si, _("%s: VOP: %u, HOP: %u, AOP: %u, SOP: %u, Other: %u"),
+					chan, vopcnt, hopcnt, aopcnt, sopcnt, othercnt);
+	}
 	snprintf(str, sizeof str, "%s: ", chan);
 	for (i = 0; i < ARRAY_SIZE(chanacs_flags); i++)
 	{
 		if (!(ca_all & chanacs_flags[i].value))
 			continue;
+		if (chanacs_flags[i].value == CA_AKICK && !show_akicks)
```

Comment:
What about entries that have `+b` *and* any other flag (except `+e`)? Those are still enforced as akicks, but would be listed here. I suppose that would be a valid way for channel operators to denote which akicks are supposed to be public (e.g. `+Ab`), but it seems a bit icky.

This is because the way akicks in general are implemented is less than ideal, but we already knew that much.

Reply:
There's some precedent for this in `chanserv/access`, which already skips over entries that are plain `+b` while handling other entires with `+b` normally. (That said, you could argue that it should also process the entry as though it didn't have the `+b` – though that'll bring a number of complications too, so that's perhaps a question for another time.)

Rethinking it, I'd tend to agree with you, though.

> (except `+e`)

Not sure `+be` should be treated specially any more than, say, `+bo` for this purpose; we do show entries with just plain `+e`, after all. (There might be an argument to be made for them to be private, analogous to ircd-side `+e`, but I haven't seen any demand for that.)

Reply:
To be clear, I wasn't arguing for `+be` to be treated specially; rather, pointing out that a `+be` entry isn't enforced.

If there's already precedent for this behaviour I suppose that's fine.